dojo/tools/openvas_xml/parser.py

```diff
@@ -0,0 +1,69 @@
+from xml.dom import NamespaceErr
+from defusedxml import ElementTree as ET
+from dojo.models import Finding
+
+
+class OpenVASXMLParser(object):
+    def get_scan_types(self):
+        return ["OpenVAS XML"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type  # no custom label for now
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import XML output of Greenbone OpenVAS XML report."
+
+    def convert_cvss_score(self, raw_value):
+        val = float(raw_value)
+        if val == 0.0:
+            return "Info"
+        elif val < 4.0:
+            return "Low"
+        elif val < 7.0:
+            return "Medium"
+        elif val < 9.0:
+            return "High"
+        else:
+            return "Critical"
+
+    def get_findings(self, file, test):
+        findings = []
+        tree = ET.parse(file)
+        root = tree.getroot()
+        if "report" not in root.tag:
+            raise NamespaceErr(
+                "This doesn't seem to be a valid Greenbone OpenVAS xml file."
+            )
+        report = root.find("report")
+        results = report.find("results")
+        for result in results:
+            for finding in result:
+                print(finding)
+                if finding.tag == "name":
```

Comment:
This is my personal preference, but when I have more than three cases to match on, I find using `match` to be easier to understand

Reply:
For me both would be fine. If the maintainers do not like it my way and therefore will not accept the PR, I can do it.

Reply:
I have the first approval, so I am closing this.